---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# SeffCovar

<!-- badges: start -->
<!-- badges: end -->

The SeffCovar R package constructs covariates for slide-effect-adjustment in epigenome-wide association studies or integrative methylation analyses.


## Installation

You can install SeffCovar using:

```{r install, eval=FALSE}
library(devtools)
install_github("julianhecker/SeffCovar")
```

## Sets S_0_20, S_20_40, S_40_60, S_60_80, S_80_100, and S_high, as described in Hecker et al. 2023 [1].

```{r example_assess}
library(SeffCovar)
length(S_high)
head(S_80_100)
```


## Approach 1: get_slide_effects_covariates

This function extracts the DNA methylation values for CpGs in S_high and performs a singular value decomposition, after scaling and centering. The resulting vectors are the PCs that can be used as covariates for slide effect adjustment.

```{r approach1}
methylation_matrix=matrix(rnorm(length(S_high)*100), nrow=length(S_high), ncol=100)
rownames(methylation_matrix)=S_high
slide_covars=get_slide_effects_covariates(methylation_matrix=methylation_matrix, input_set=S_high)
head(slide_covars)
```


## Approach 2: get_ComBat_slide_effect_covariates


This function extracts the DNA methylation values for CpGs in S_high and performs part of the ComBat adjustment [2]. The function uses the estimated slide effect gamma's and performs a singular value decomposition on the resulting matrix. Again, this leads to PCs that can be used as covariates for slide effect adjustment. The functions is based on a modification of the ComBat function in the sva package [3]. The original code for ComBat from the sva package that can be found at
https://bioconductor.org/packages/release/bioc/html/sva.html.

```{r approach2}
methylation_matrix=matrix(rnorm(length(S_high)*100), nrow=length(S_high), ncol=100)
rownames(methylation_matrix)=S_high
slide=rep(paste0("slide",1:10),10)
slide_covars=get_ComBat_slide_effect_covariates(methylation_matrix=methylation_matrix, slide=slide, mod=NULL, input_set=S_high)
head(slide_covars)
```


# References

[1] A consistent pattern of slide effects in Illumina DNA methylation BeadChip array data.
Hecker J, Lee S, Kachroo P, Prokopenko D, Maaser-Hecker A, Lutz SM, Hahn G, Irizarry R, Weiss ST, DeMeo DL, Lange C.
Epigenetics. 2023 Dec;18(1):2257437. doi: 10.1080/15592294.2023.2257437. Epub 2023 Sep 20.
PMID: 37731367 

[2] Adjusting batch effects in microarray
expression data using empirical Bayes methods. Johnson WE, Li C, Rabinovic A. Biostatistics. 2007
Jan;8(1):118-27. doi: 10.1093/biostatistics/kxj037. Epub 2006 Apr 21. PMID:
16632515

[3] The sva package for removing batch effects and other unwanted variation in high-throughput experiments.
Leek JT, Johnson WE, Parker HS, Jaffe AE, Storey JD.
Bioinformatics. 2012 Mar 15;28(6):882-3. doi: 10.1093/bioinformatics/bts034. Epub 2012 Jan 17.
PMID: 22257669

